internal/clientconn/conn.go

```diff
@@ -198,7 +198,43 @@ func (c *conn) run(ctx context.Context) (err error) {
 	for {
 		var reqHeader *wire.MsgHeader
 		var reqBody wire.MsgBody
+		var resHeader *wire.MsgHeader
+		var resBody wire.MsgBody
+		var validationErr *wire.ValidationError
+
 		reqHeader, reqBody, err = wire.ReadMessage(bufr)
+		if err != nil && errors.As(err, &validationErr) {
+			res := new(wire.OpMsg)
+
+			must.NoError(res.SetSections(wire.OpMsgSection{
+				Documents: []*types.Document{validationErr.Document()},
+			}))
+
+			var b []byte
+
+			b, err = res.MarshalBinary()
+			if err != nil {
+				panic(err)
+			}
+
+			resHeader = &wire.MsgHeader{
+				OpCode:        reqHeader.OpCode,
+				RequestID:     atomic.AddInt32(&c.lastRequestID, 1),
+				ResponseTo:    reqHeader.RequestID,
+				MessageLength: int32(wire.MsgHeaderLen + len(b)),
+			}
```

Comment:
Now we have two places where we make `resHeader`: here and in `route`. I'm not sure if it's a good solution, especially, having in mind that there are some `TODOs` around resHeader in `route`. Should we move both of them to a separate function that would make `resHeader`, or is it too much? (Maybe someone else has a better idea...)

Reply:
I tried to move it somewhere but the result wasn’t looking good. I suggest work on that later.